apps/frontend/src/components/listingCard.tsx

```diff
@@ -0,0 +1,146 @@
+import { Grid, Button, Typography } from "@mui/material";
+import PropTypes from "prop-types";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import _axios_instance from "../_axios_instance.tsx";
+
+interface ListingObject {
+  listingID: string;
+  sellerID: string;
+  sellerName: string;
+  title: string;
+  description: string;
+  price: number;
+  dateCreated: string;
+  imageUrl: string;
+}
+
+interface ListingCardProps {
+  searchPerformed: boolean;
+  listing: ListingObject;
+}
+
+const ListingCard = ({ searchPerformed, listing }: ListingCardProps) => {
+  const navigate = useNavigate();
+  const [isGreyedOut, setIsGreyedOut] = useState(false);
+
+  const handleListingClick = () => {
+    navigate(`/listing/${listing.listingID}`);
+  };
+
+  const priceFormatter = new Intl.NumberFormat("en-CA", {
+    style: "currency",
+    currency: "CAD",
+  });
+
+  const convertDate = (dateString: string) => {
+    const dateTimeObject = new Date(dateString);
+    return dateTimeObject.toDateString();
+  };
+
+  const handleNotInterested = () => {
```

Comment:
It seems that the https://whateverthiswillbe.com/api/recommendations/stop is being called quite a few times in the tests(specifically the _"Test not interested button"_ test), this may be worth mocking up and testing specifically as duplicating this call will probably do bad things to the recommendations and we'll be marked based on duplicate API calls for code quality.

You can probably fix this similarly to how it was fixed with the global variable for recording its state.

![image](https://github.com/UVicMartletplace/martletplace/assets/54737724/6ef7712b-8d0a-49b1-9d5b-9215a33ca17d)